test/util/test_runners/test_runner_args.py

```diff
@@ -0,0 +1,47 @@
+from argparse import ArgumentParser
+
+
+class TestRunnerArgs:
+    """
+    DTO to hold the tests environment arguments
+    """
+
+    def __init__(self, hostname, account, username, password, invoke_cli_as_process,
+                 working_cli_path_only_for_init: str,
+                 cli_to_test_path: str, files_folder: str, test_name_identifier="integration"):
+        self.hostname = hostname
+        self.account = account
+        self.username = username
+        self.password = password
+        self.test_name_identifier = test_name_identifier
+        self.invoke_cli_as_process = invoke_cli_as_process
+        self.working_cli_path_only_for_init = working_cli_path_only_for_init
+        self.cli_to_test_path = cli_to_test_path
+        self.files_folder = files_folder
+        # will throw if the user test the same known working binaries
+        if cli_to_test_path is not None and working_cli_path_only_for_init.lower() == cli_to_test_path.lower():
+            raise RuntimeError("cli to start enviroment and tested cli are on the same path")
+
+    @staticmethod
+    def create_from_args():
+        parser = ArgumentParser(description='Test arguments')
```

Comment:
I would even add a docstring with examples of how to run the two flows

""""
Method to accept input from the user on how to run the test suite. 
There are two flows:
1. Running as a process (executable). This option can be run cross-platform
Example: ...
2. Running as code (requiring Python on the machine)
Example: ...

"""

Reply:
copied from what you wrote in the contrib.md